packages/frontend/apps/electron/src/main/windows-manager/main-window.ts

```diff
@@ -116,11 +116,17 @@ export class MainWindowManager {
       uiSubjects.onFullScreen$.next(mainWindow.isFullScreen());
     });
 
+    app.on('before-quit', () => {
```

Comment:
I think it's good to have a global cleanup registry instead of register everywhere